test/component_allocations_test.go

```diff
@@ -0,0 +1,197 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/maxio-com/ab-golang-sdk/errors"
+	"github.com/maxio-com/ab-golang-sdk/models"
+	"github.com/stretchr/testify/suite"
+)
+
+type ComponentAlocationSuite struct {
+	APISuite
+}
+
+func TestComponentSuite(t *testing.T) {
+	suite.Run(t, new(ComponentAlocationSuite))
+}
+
+type OnOffComponent struct {
+	Name                      string       `json:"name"`
+	Prices                    []Price      `json:"prices"`
+	PricingScheme             string       `json:"pricing_scheme"`
+	AllowFractionalQuantities bool         `json:"allow_fractional_quantities"`
+	PricePoint                []PricePoint `json:"price_points"`
+	UnitPrice                 string       `json:"unit_price"`
+}
+
+type PricePoint struct {
+	Name          string  `json:"name"`
+	Prices        []Price `json:"prices"`
+	PricingScheme string  `json:"pricing_scheme,omitempty"`
+}
+
+type OneOfBody struct {
+	OneOfComponent OnOffComponent `json:"on_off_component"`
+}
+
+type QuantityBasedComponent struct {
+	Name                      string       `json:"name"`
+	UnitName                  string       `json:"unit_name"`
+	PricingScheme             string       `json:"pricing_scheme"`
+	AllowFractionalQuantities bool         `json:"allow_fractional_quantities"`
+	PricePoint                []PricePoint `json:"price_points"`
+	UnitPrice                 string       `json:"unit_price"`
+}
+
+type QuantityBody struct {
+	QuantityBasedComponent QuantityBasedComponent `json:"quantity_based_component"`
+}
+
+func (s *ComponentAlocationSuite) TestComponentAllocations() {
+	ctx := context.Background()
+
+	customer := s.createCustomer(ctx)
+	pf := s.createProductFamily(ctx)
+	product := s.createProduct(ctx, *pf.Id)
+	subscription := s.newSubscription(customer, product, "", []models.CreateSubscriptionComponent{})
+	subResp, err := s.client.SubscriptionsController().CreateSubscription(ctx, &models.CreateSubscriptionRequest{Subscription: subscription})
+	s.NoError(err)
+	s.Equal(http.StatusCreated, subResp.Response.StatusCode)
+
+	onOffResp, err := s.client.ComponentsController().CreateComponent(
+		ctx,
+		*pf.Id,
+		models.ComponentKindPath_ONOFFCOMPONENTS,
+		interfacePtr(OneOfBody{
+			OneOfComponent: OnOffComponent{
+				Name:                      "One of component",
+				AllowFractionalQuantities: true,
```

Comment:
could we remove `AllowFractionalQuantities` from `OneOfComponent`, and then assert for integer in allocation responses for on/off components?
`s.Equal(fmt.Sprintf("%.1f", expected[0].Quantity), *prevAllocations[0].Quantity)`

Reply:
it sends back floatType but without any tenths of the number

Reply:
weird, it should be int.. Thanks for verifying!